psycopg/psycopg/errors.py

```diff
@@ -544,15 +548,9 @@ def lookup(sqlstate: str) -> type[Error]:
 
 
 def error_from_result(result: PGresult, encoding: str = "utf-8") -> Error:
-    from psycopg import pq
-
     state = result.error_field(DiagnosticField.SQLSTATE) or b""
-    cls = _class_for_state(state.decode("ascii"))
-    return cls(
-        pq.error_message(result, encoding=encoding),
-        info=result,
-        encoding=encoding,
-    )
+    cls = _class_for_state(state.decode("utf-8", "replace"))
```

Comment:
```suggestion
    cls = _class_for_state(state.decode(encoding, "replace"))
```
? was "utf-8" here intended?

Reply:
This value is only supposed to be a 5 chars sqlstate codes, and any encoding would do to decode it. It was "ascii" before, and would have failed in case for networking reason we end up with random characters here, because there is no "replace".

So I thought, instead of failing with a decoding error here, let's keep going down the rabbit hole and see what get raised, as it might be more informative (`_class_for_state()` would return `DatabaseError` in case you throw random values at it).